test/functional/sbom/test_cycloneDX.py

```diff
@@ -0,0 +1,139 @@
+import textwrap
+
+import pytest
+
+from conan.test.assets.genconanfile import GenConanfile
+from conan.test.utils.tools import TestClient
+from conans.util.files import save
+import os
+
+# Using the sbom tool with "conan create"
+sbom_hook_post_package = """
+import json
+import os
+from conan.errors import ConanException
+from conan.api.output import ConanOutput
+from conan.tools.sbom.cycloneDX import cyclonedx_1_4
+
+def post_package(conanfile):
+    try:
+        sbom_cyclonedx_1_4 = cyclonedx_1_4(conanfile.subgraph)
+        metadata_folder = conanfile.package_metadata_folder
+        file_name = "cyclonedx_1_4.json"
+        with open(os.path.join(metadata_folder, file_name), 'w') as f:
+            json.dump(sbom_cyclonedx_1_4, f, indent=4)
+        ConanOutput().success(f"CYCLONEDX CREATED - {conanfile.package_metadata_folder}")
+    except Exception as e:
+        ConanException("error generating CYCLONEDX file")
```

Comment:
Is this really necessary? If something fails in the hook, then it makes debugging it more difficult?
Don't the hooks have exception capturing?